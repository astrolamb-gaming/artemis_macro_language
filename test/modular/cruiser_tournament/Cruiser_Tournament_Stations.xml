<mission_data>
    <values>
        <range name="stations">
            <value name="DS1" nuke="0" homing="7" EMP="0" mine="1" Pshock="2" type="Science" x="69500" z="69000" bonus="Vigoranium Nodules"/>
            <value name="DS2" nuke="1" homing="10" EMP="0" mine="1" Pshock="2" type="Civilian" x="44000" z="71000" bonus="High Density Power Cells" />
            <value name="DS3" nuke="0" homing="27" EMP="0" mine="1" Pshock="2" type="Science" x="76000" z="95000" bonus="Carapaction Coils" />
            <value name="DS4" nuke="0" homing="27" EMP="0" mine="1" Pshock="2" type="Civilian" x="33600" z="92500" bonus="Tauron Focusers" />
        </range>
    </values>
    <start>
        <repeat range="stations">
            <clear_player_station_carried name="${name}"/>
            <set_player_station_carried name="${name}" singleSeatName="Pilgrim" raceKeys="TSN player" hullKeys="singleseat shuttle"/>

            <create type="station" x="${x}" y="0.0" z="${z}" angle="0" name="${name}" raceKeys="Terran friendly" hullKeys="${type} Base"/>
            <set_object_property property="missileStoresNuke" value="${nuke}" name="${name}"/>
            <set_object_property property="missileStoresHoming" value="${homing}" name="${name}"/>
            <set_object_property property="missileStoresEMP" value="${EMP}" name="${name}"/>
            <set_object_property property="missileStoresMine" value="${mine}" name="${name}"/>
            <set_object_property property="missileStoresPshock" value="${Pshock}" name="${name}"/>
        </repeat>
    </start>
    <!-- Hints from Bases -->
    <repeat range="stations">
        <event name="${name} Hint">
            <if_variable name="${name}_hint" comparator="EQUALS" value="0"/>
            <if_docked name="${name}"/>
            <incoming_comms_text from="${name}" type="STATION">
    Bonus available! If you collect all ${bonus} Nodules a bonus enemy fleet will attack. You can score extra tonnage by destroying it.
            </incoming_comms_text>
            <set_variable name="${name}_hint" value="1"/>
        </event>
        <event name="${name} Destroyed">
            <if_variable name="${name}_dead" comparator="EQUALS" value="0"/>
            <if_not_exists name="${name}"/>
            <big_message title="Friendly Base ${name} Destroyed" subtitle1="Penalty -90 kilotons" subtitle2=""/>
            <log text="Friendly Base ${name} Destroyed"/>
            <set_variable name="${name}_dead" value="1"/>
            <set_variable name="Tonnage" value="Tonnage-90"/>
            <set_variable name="report_score" value="1"/>
            <set_timer name="report_timer" seconds="8"/>
        </event>
    </repeat>
</mission_data>