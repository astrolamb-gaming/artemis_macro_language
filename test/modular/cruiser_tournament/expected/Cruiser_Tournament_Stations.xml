<mission_data>
  <start>
    <clear_player_station_carried name="DS1"/>
    <set_player_station_carried name="DS1" singleSeatName="Pilgrim" raceKeys="TSN player" hullKeys="singleseat shuttle"/>
    <create type="station" x="69500" y="0.0" z="69000" angle="0" name="DS1" raceKeys="Terran friendly" hullKeys="Science Base"/>
    <set_object_property property="missileStoresNuke" value="0" name="DS1"/>
    <set_object_property property="missileStoresHoming" value="7" name="DS1"/>
    <set_object_property property="missileStoresEMP" value="0" name="DS1"/>
    <set_object_property property="missileStoresMine" value="1" name="DS1"/>
    <set_object_property property="missileStoresPshock" value="2" name="DS1"/>
    <clear_player_station_carried name="DS2"/>
    <set_player_station_carried name="DS2" singleSeatName="Pilgrim" raceKeys="TSN player" hullKeys="singleseat shuttle"/>
    <create type="station" x="44000" y="0.0" z="71000" angle="0" name="DS2" raceKeys="Terran friendly" hullKeys="Civilian Base"/>
    <set_object_property property="missileStoresNuke" value="1" name="DS2"/>
    <set_object_property property="missileStoresHoming" value="10" name="DS2"/>
    <set_object_property property="missileStoresEMP" value="0" name="DS2"/>
    <set_object_property property="missileStoresMine" value="1" name="DS2"/>
    <set_object_property property="missileStoresPshock" value="2" name="DS2"/>
    <clear_player_station_carried name="DS3"/>
    <set_player_station_carried name="DS3" singleSeatName="Pilgrim" raceKeys="TSN player" hullKeys="singleseat shuttle"/>
    <create type="station" x="76000" y="0.0" z="95000" angle="0" name="DS3" raceKeys="Terran friendly" hullKeys="Science Base"/>
    <set_object_property property="missileStoresNuke" value="0" name="DS3"/>
    <set_object_property property="missileStoresHoming" value="27" name="DS3"/>
    <set_object_property property="missileStoresEMP" value="0" name="DS3"/>
    <set_object_property property="missileStoresMine" value="1" name="DS3"/>
    <set_object_property property="missileStoresPshock" value="2" name="DS3"/>
    <clear_player_station_carried name="DS4"/>
    <set_player_station_carried name="DS4" singleSeatName="Pilgrim" raceKeys="TSN player" hullKeys="singleseat shuttle"/>
    <create type="station" x="33600" y="0.0" z="92500" angle="0" name="DS4" raceKeys="Terran friendly" hullKeys="Civilian Base"/>
    <set_object_property property="missileStoresNuke" value="0" name="DS4"/>
    <set_object_property property="missileStoresHoming" value="27" name="DS4"/>
    <set_object_property property="missileStoresEMP" value="0" name="DS4"/>
    <set_object_property property="missileStoresMine" value="1" name="DS4"/>
    <set_object_property property="missileStoresPshock" value="2" name="DS4"/>
    <set_comms_button text="Request Cargo Report" sideValue="2" player_slot="0"/>
  </start>
  
  <event name="DS1 Hint">
    <if_variable name="DS1_hint" comparator="EQUALS" value="0"/>
    <if_docked name="DS1"/>
    <incoming_comms_text from="DS1" type="STATION">
  Bonus available! If you collect all Vigoranium Nodules a bonus enemy fleet will attack. You can score extra tonnage by destroying it.
      </incoming_comms_text>
    <set_variable name="DS1_hint" value="1"/>
  </event>
  <event name="DS1 Destroyed">
    <if_variable name="DS1_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="DS1"/>
    <big_message title="Friendly Base DS1 Destroyed" subtitle1="Penalty -90 kilotons" subtitle2=""/>
    <log text="Friendly Base DS1 Destroyed"/>
    <set_variable name="DS1_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage-90"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
  </event>
  <event name="Instructions from DS1">
    <if_variable name="Minutes" comparator="EQUALS" value="47"/>
    <if_variable name="Message" comparator="EQUALS" value="0"/>
    <incoming_comms_text from="DS1" type="STATION">
    Our base can produces Xiridium. We need Platinum to make Nukes. If you send us Salistra or Augite we can make EMP torpedoes.
      </incoming_comms_text>
    <set_variable name="Message" value="1"/>
  </event>
  <event name="DS2 Hint">
    <if_variable name="DS2_hint" comparator="EQUALS" value="0"/>
    <if_docked name="DS2"/>
    <incoming_comms_text from="DS2" type="STATION">
  Bonus available! If you collect all High Density Power Cells Nodules a bonus enemy fleet will attack. You can score extra tonnage by destroying it.
      </incoming_comms_text>
    <set_variable name="DS2_hint" value="1"/>
  </event>
  <event name="DS2 Destroyed">
    <if_variable name="DS2_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="DS2"/>
    <big_message title="Friendly Base DS2 Destroyed" subtitle1="Penalty -90 kilotons" subtitle2=""/>
    <log text="Friendly Base DS2 Destroyed"/>
    <set_variable name="DS2_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage-90"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
  </event>
  <event name="Instructions from DS2">
    <if_variable name="Minutes" comparator="EQUALS" value="46"/>
    <if_variable name="Message" comparator="EQUALS" value="1"/>
    <incoming_comms_text from="DS2" type="STATION">
    Our base can produces Salistra. We need Xiridium to make Nukes. If you send us Platinum or Augite we can make EMP torpedoes.
      </incoming_comms_text>
    <set_variable name="Message" value="2"/>
  </event>
  <event name="DS3 Hint">
    <if_variable name="DS3_hint" comparator="EQUALS" value="0"/>
    <if_docked name="DS3"/>
    <incoming_comms_text from="DS3" type="STATION">
  Bonus available! If you collect all Carapaction Coils Nodules a bonus enemy fleet will attack. You can score extra tonnage by destroying it.
      </incoming_comms_text>
    <set_variable name="DS3_hint" value="1"/>
  </event>
  <event name="DS3 Destroyed">
    <if_variable name="DS3_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="DS3"/>
    <big_message title="Friendly Base DS3 Destroyed" subtitle1="Penalty -90 kilotons" subtitle2=""/>
    <log text="Friendly Base DS3 Destroyed"/>
    <set_variable name="DS3_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage-90"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
  </event>
  <event name="Instructions from DS3">
    <if_variable name="Minutes" comparator="EQUALS" value="45"/>
    <if_variable name="Message" comparator="EQUALS" value="2"/>
    <incoming_comms_text from="DS3" type="STATION">
    Our base can produces Augite. We need Salistra to make Nukes. If you send us Xiridium or Platinum we can make EMP torpedoes.
      </incoming_comms_text>
    <set_variable name="Message" value="3"/>
  </event>
  <event name="DS4 Hint">
    <if_variable name="DS4_hint" comparator="EQUALS" value="0"/>
    <if_docked name="DS4"/>
    <incoming_comms_text from="DS4" type="STATION">
  Bonus available! If you collect all Tauron Focusers Nodules a bonus enemy fleet will attack. You can score extra tonnage by destroying it.
      </incoming_comms_text>
    <set_variable name="DS4_hint" value="1"/>
  </event>
  <event name="DS4 Destroyed">
    <if_variable name="DS4_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="DS4"/>
    <big_message title="Friendly Base DS4 Destroyed" subtitle1="Penalty -90 kilotons" subtitle2=""/>
    <log text="Friendly Base DS4 Destroyed"/>
    <set_variable name="DS4_dead" value="1"/>
    <set_variable name="Tonnage" value="Tonnage-90"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
  </event>
  <event name="Instructions from DS4">
    <if_variable name="Minutes" comparator="EQUALS" value="44"/>
    <if_variable name="Message" comparator="EQUALS" value="3"/>
    <incoming_comms_text from="DS4" type="STATION">
    Our base can produces Platinum. We need Augite to make Nukes. If you send us Xiridium or Salistra we can make EMP torpedoes.
      </incoming_comms_text>
    <set_variable name="Message" value="4"/>
  </event>
  <event name="Instructions from Scorekeeper">
    <if_comms_button text="Request Cargo Report"/>
    <incoming_comms_text from="Scorekeeper" type="ALERT">
      DS1 makes Xiridium and prefers Platinum
      </incoming_comms_text>
    <incoming_comms_text from="Scorekeeper" type="ALERT">
      DS2 makes Salistra and prefers Xiridium
      </incoming_comms_text>
    <incoming_comms_text from="Scorekeeper" type="ALERT">
      DS3 makes Augite and prefers Salistra
      </incoming_comms_text>
    <incoming_comms_text from="Scorekeeper" type="ALERT">
      DS4 makes Platinum and prefers Augite
      </incoming_comms_text>
    <incoming_comms_text from="Scorekeeper" type="ALERT">
      Any base will accept any cargo it doesn't produce.
    </incoming_comms_text>
  </event>
</mission_data>